src/main/java/com/achdev/onlinebookstoreapp/controller/CategoryController.java

```diff
@@ -0,0 +1,186 @@
+package com.achdev.onlinebookstoreapp.controller;
+
+import com.achdev.onlinebookstoreapp.dto.book.BookDtoWithoutCategoryIds;
+import com.achdev.onlinebookstoreapp.dto.category.CategoryDto;
+import com.achdev.onlinebookstoreapp.dto.category.CreateCategoryRequestDto;
+import com.achdev.onlinebookstoreapp.dto.errors.CommonApiErrorResponse;
+import com.achdev.onlinebookstoreapp.dto.page.PageResponse;
+import com.achdev.onlinebookstoreapp.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories management", description = "Endpoints for managing categories")
+@RestController
+@RequestMapping("/categories")
+@RequiredArgsConstructor
+public class CategoryController {
+    private static final String RESPONSE_CODE_OK = "200";
```

Comment:
If you have similar constants in each controller, it's better to replace it to util class (for example)

Reply:
Ok